### PR TITLE
Update BBEs with default payload param support

### PR DESCRIPTION
**Pull request by chamil321**

### CODE REVIEW

**Comment by TharmiganK** [comment]
The changes are not yet released, so the build will fail. Can merge this once the `2201.5.0` is released.

File: examples/http-query-parameter/http_query_parameter.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
...
+The query parameter in the resource argument represents the query segment of the request URL. The argument name should be the key of the query, and its value is mapped during the runtime by extracting it from the URL. Except for resolving ambiguity with the default payload parameter, no additional annotation is required for the query parameter. However, the `@http:Query` annotation can be used in such cases. The supported types are `string`, `int`, `float`, `boolean`, `decimal`, `map<json>`, and `array` types of the aforementioned types. The query parameter type can be nilable (e.g., (`string? bar`)) and defaultable (e.g., (`string bar = "hello"`)). When a request contains query segments, retrieving them as resource arguments is much simpler and well-recommended. Alternatively, the `http:Request` also provides related methods to retrieve query parameters.
```
Comment (shafreenAnfar) [inline examples/http-query-parameter/http_query_parameter.md:3]: I think we can revert these text changes. We can add it later as a note if needed. `@http:Query` is need only in extreme edge cases.

File: examples/http-service-data-binding/http_service_data_binding.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
+HTTP service payload data binding allows accessing the request payload using a resource signature parameter. The resource parameter type should be of `anydata` subtype. The default payload parameter types are basic structured types such as `map`, `array`, `tuple`, `table`, `record` and `xml` which is a sequence type, which can be defined in the signature without any annotation. For other types, the `@http:Payload` annotation is required. If the signature includes more than one structured type, the `@http:Payload` annotation should be used to resolve any ambiguity. The default payload parameter support is limited to `POST`, `PUT`, `PATCH`, `DELETE`, and `DEFAULT` accessors. If the data binding process fails, the client receives a 400 Bad Request response. This feature allows direct access to the request payload from the resource.
```
Comment (shafreenAnfar) [inline examples/http-service-data-binding/http_service_data_binding.md:3]: HTTP service payload data binding allows accessing the request payload using a resource signature parameter. The resource parameter type should be of `anydata` subtype. By default parameters with type `map `, `array`, `tuple`, `table`, `record` and `xml` are mapped to the payload. For other types, the `@http:Payload` annotation is required and If the signature includes more than one of the aforementioned types, the `@http:Payload` should be used to resolve the ambiguity. This behaviour is limited to `POST`, `PUT`, `PATCH`, `DELETE`, and `DEFAULT` accessors.  If the data binding process fails, the client receives a 400 Bad Request response. This feature allows direct access to the request payload from the resource.
```suggestion
HTTP service payload data binding allows accessing the request payload using a resource signature parameter. The resource parameter type should be of `anydata` subtype. By default parameters with type `map `, `array`, `tuple`, `table`, `record` and `xml` are mapped to the payload. For other types, the `@http:Payload` annotation is required and If the signature includes more than one of the aforementioned types, the `@http:Payload` should be used to resolve the ambiguity. This behaviour is limited to `POST`, `PUT`, `PATCH`, `DELETE`, and `DEFAULT` accessors.  If the data binding process fails, the client receives a 400 Bad Request response. This feature allows direct access to the request payload from the resource.
```

File: examples/http-query-parameter/http_query_parameter.md (hunk 1 above)
Comment (TharmiganK) [inline examples/http-query-parameter/http_query_parameter.md:3]:
```suggestion
The query parameter in the resource argument represents the query segment of the request URL. The argument name should be the key of the query, and its value is mapped during the runtime by extracting it from the URL. The query parameter does not need any additional annotation. The supported types are `string`, `int`, `float`, `boolean`, `decimal`, and `array` types of the aforementioned types. The query parameter type can be nilable (e.g., (`string? bar`)) and defaultable (e.g., (`string bar = "hello"`)). When a request contains query segments, retrieving them as resource arguments is much simpler and well-recommended. Alternatively, the `http:Request` also provides related methods to retrieve query parameters.
```

Comment (TharmiganK) [reply examples/http-query-parameter/http_query_parameter.md]: Fixed

Comment (TharmiganK) [reply examples/http-query-parameter/http_query_parameter.md]: Fixed via [fcd7eff](https://github.com/ballerina-platform/ballerina-distribution/pull/4119/commits/fcd7eff8fab4660b16bfcb622b4bbf9504761107)

File: examples/http-service-data-binding/http_service_data_binding.md (hunk 2 above)
Comment (TharmiganK) [reply examples/http-service-data-binding/http_service_data_binding.md]: Fixed via [fcd7eff](https://github.com/ballerina-platform/ballerina-distribution/pull/4119/commits/fcd7eff8fab4660b16bfcb622b4bbf9504761107)

**Review by shafreenAnfar — DISMISSED** [review]

File: examples/http-send-different-status-codes-with-payload/http_send_different_status_codes_with_payload.bal [hunk 3]
```diff
@@ -21,7 +21,7 @@ table<Album> key(title) albums = table [
 service / on new http:Listener(9090) {
 
     // The resource returns `409 Conflict` status code as the error response status code using build in StatusCodeResponse.
```
Comment (praneesha) [inline examples/http-send-different-status-codes-with-payload/http_send_different_status_codes_with_payload.bal:23]:
```suggestion
    // The resource returns the `409 Conflict` status code as the error response status code using the build-in `StatusCodeResponse`.
```
Is it `build-in` or `built-in`?

File: examples/http-send-different-status-codes/http_send_different_status_codes.bal [hunk 4]
```diff
@@ -14,7 +14,7 @@ service / on new http:Listener(9090) {
 
     // The resource returns `409 Conflict` status code as the error response status code using
```
Comment (praneesha) [inline examples/http-send-different-status-codes/http_send_different_status_codes.bal:15]:
```suggestion
    // The resource returns the `409 Conflict` status code as the error response status code using 
```

File: examples/http-send-different-status-codes/http_send_different_status_codes.bal [hunk 5]
```diff
@@ -14,7 +14,7 @@ service / on new http:Listener(9090) {
 
     // The resource returns `409 Conflict` status code as the error response status code using 
     // `StatusCodeResponse` constants. This constant does not have a body or headers.
```
Comment (praneesha) [inline examples/http-send-different-status-codes/http_send_different_status_codes.bal:16]:
```suggestion
    // the `StatusCodeResponse` constants. This constant does not have a body or headers.
```

File: examples/http-service-data-binding/http_service_data_binding.md [hunk 6]
```diff
@@ -1,6 +1,6 @@
 # REST service - Payload data binding
 
-HTTP service payload data binding allows accessing the request payload using a resource signature parameter. The resource parameter which is bound to the request payload should be annotated with `@http:Payload` annotation. The resource parameter type should be `anydata`. If the data binding fails, a `400 Bad Request` response is sent to the client. Use this to access the request payload directly from the resource.
+HTTP service payload data binding allows accessing the request payload using a resource signature parameter. The resource parameter type should be of `anydata` subtype. By default parameters with type `map `, `array`, `tuple`, `table`, `record` and `xml` are mapped to the payload. For other types, the `@http:Payload` annotation is required and If the signature includes more than one of the aforementioned types, the `@http:Payload` should be used to resolve the ambiguity. This behaviour is limited to `POST`, `PUT`, `PATCH`, `DELETE`, and `DEFAULT` accessors.  If the data binding process fails, the client receives a 400 Bad Request response. This feature allows direct access to the request payload from the resource.
```
Comment (praneesha) [inline examples/http-service-data-binding/http_service_data_binding.md:3]:
```suggestion
HTTP service payload data binding allows accessing the request payload using a resource signature parameter. The resource parameter type should be a sub type of `anydata`. By default, parameters with the `map `, `array`, `tuple`, `table`, `record` and `xml` types are mapped to the payload. For other types, the `@http:Payload` annotation is required and If the signature includes more than one of the aforementioned types, the `@http:Payload` should be used to resolve the ambiguity. This behaviour is limited to the `POST`, `PUT`, `PATCH`, `DELETE`, and `DEFAULT` accessors.  If the data binding process fails, the client receives a `400 Bad Request` response. This feature allows direct access to the request payload from the resource.
```

File: examples/mysql-atomic-transaction/mysql_atomic_transaction.bal [hunk 7]
```diff
@@ -20,7 +20,7 @@ service / on new http:Listener(8080) {
         self.db = check new ("localhost", "root", "Test@123", "MUSIC_STORE", 3306);
     }
 
-    resource function post 'order(@http:Payload Order salesOrder) returns http:Created|error {
+    resource function post 'order(Order salesOrder) returns http:Created|error {
         transaction {
             // Insert into `sales_order` table.
```
Comment (praneesha) [inline examples/mysql-atomic-transaction/mysql_atomic_transaction.bal:25]:
```suggestion
            // Insert into the `sales_order` table.
```

File: examples/rabbitmq-client-basic-auth/rabbitmq_client_basic_auth.bal [hunk 8]
```diff
@@ -24,7 +24,7 @@ service / on new http:Listener(9092) {
         );
     }
 
-    resource function post orders(@http:Payload Order newOrder) returns http:Accepted|error {
+    resource function post orders(Order newOrder) returns http:Accepted|error {
         // Publishes the message using newClient and the routing key named OrderQueue.
```
Comment (praneesha) [inline examples/rabbitmq-client-basic-auth/rabbitmq_client_basic_auth.bal:28]:
```suggestion
        // Publishes the message using the `newClient` and the routing key named `OrderQueue`.
```

File: examples/rabbitmq-client-secure-connection/rabbitmq_client_secure_connection.bal [hunk 9]
```diff
@@ -23,7 +23,7 @@ service / on new http:Listener(9092) {
         );
     }
 
-    resource function post orders(@http:Payload Order newOrder) returns http:Accepted|error {
+    resource function post orders(Order newOrder) returns http:Accepted|error {
         // Publishes the message using newClient and the routing key named OrderQueue.
```
Comment (praneesha) [inline examples/rabbitmq-client-secure-connection/rabbitmq_client_secure_connection.bal:27]:
```suggestion
        // Publishes the message using the `newClient` and the routing key named `OrderQueue`.
```

File: examples/rabbitmq-producer/rabbitmq_producer.bal [hunk 10]
```diff
@@ -17,7 +17,7 @@ service / on new http:Listener(9092) {
         self.orderClient = check new (rabbitmq:DEFAULT_HOST, rabbitmq:DEFAULT_PORT);
     }
 
-    resource function post orders(@http:Payload Order newOrder) returns http:Accepted|error {
+    resource function post orders(Order newOrder) returns http:Accepted|error {
         // Publishes the message using newClient and the routing key named OrderQueue.
```
Comment (praneesha) [inline examples/rabbitmq-producer/rabbitmq_producer.bal:21]:
```suggestion
        // Publishes the message using the `newClient` and the routing key named `OrderQueue`.
```

File: examples/rabbitmq-transaction-producer/rabbitmq_transaction_producer.bal [hunk 11]
```diff
@@ -17,7 +17,7 @@ service / on new http:Listener(9092) {
         self.orderClient = check new (rabbitmq:DEFAULT_HOST, rabbitmq:DEFAULT_PORT);
     }
 
-    resource function post orders(@http:Payload Order newOrder) returns http:Accepted|error {
+    resource function post orders(Order newOrder) returns http:Accepted|error {
         transaction {
             // Publishes the message using newClient and the routing key named OrderQueue.
```
Comment (praneesha) [inline examples/rabbitmq-transaction-producer/rabbitmq_transaction_producer.bal:22]:
```suggestion
            // Publishes the message using the `newClient` and the routing key named `OrderQueue`.
```

File: examples/http-send-different-status-codes-with-payload/http_send_different_status_codes_with_payload.bal (hunk 3 above)
Comment (TharmiganK) [reply examples/http-send-different-status-codes-with-payload/http_send_different_status_codes_with_payload.bal:23]: Its `built-in`
```suggestion
    // The resource returns the `409 Conflict` status code as the error response status code using the built-in `StatusCodeResponse`.
```

**Review by praneesha — APPROVED** [review]

**Pull request merged**